### ISSUE TITLE
more detailed parameters

### ISSUE THREAD

**Issue by jackweiwang** [GjjvdBurg/GenSVM#3]
lambda 
kappa 
epsilon

 if n = 14000+ objects with K = 6 classes and m = 10000+ attributes

How should these parameters be adjusted?

I'm using HOG Feature and m is very large and I am training for a long time.

thank you.

**Comment by GjjvdBurg** [comment]
Thanks for checking out GenSVM!

The parameters have the following meaning:
1. lambda is the cost parameter of the regularization term. It helps to prevent overfitting. 
2. kappa is a parameter that determines how "quadratic" the errors in the hinge function: kappa = -0.999 approaches the absolute hinge used in classic SVMs and a larger value of kappa makes the hinge more and more quadratic. You can tune this value, but with such a large dataset maybe it's best to keep the default.
3. epsilon is the stopping criterion for the optimization algorithm. Basically, the smaller epsilon is, the more accurately the SVM will be fitted. You may want to try using a larger value for epsilon than the default, due to the size of your dataset.

There's no simple way to know which parameter values are "best" for your dataset. But considering the size, I suggest to start with values that give you results quickly. You can save the model using the ``-m`` command line switch. Later you can then use this model as a starting position using the ``-s`` switch. This allows you to start the model training from a better position than a random start.

I hope this helps!

**Comment by jackweiwang** [comment]
Thank you very much for your answer, I see.

**Comment by GjjvdBurg** [comment]
@jackweiwang one more thing: it may also be beneficial for speed if you scale your data to [-1, 1] using for instance a [maxabs_scale](http://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.maxabs_scale.html). This scaling doesn't destroy sparsity and it sometimes allows GenSVM to converge faster. Also, for SVMs it is common to scale the data such that all features have the same range because this ensures that the regularization term has the same effect on all features.

**Comment by jackweiwang** [comment]
ok，  My dataset has been normalized ([0, 1])， Under training......

**Comment by jackweiwang** [comment]
I'm sorry to bother you again
The classification is excellent, but I predict it slowly, so  I want change 'double' to 'float' ,Is there another problem?
![image](https://user-images.githubusercontent.com/19305790/32591523-6b810488-c55a-11e7-92ce-783d7f6fdb51.png)

**Comment by GjjvdBurg** [comment]
@jackweiwang no worries! I'm glad to hear that the classification is good. Changing double to float may help, but depending on your processor type it may be just as fast as using double ([link](https://stackoverflow.com/a/4584707/1154005)). I would recommend measuring the computation time to see if it actually makes a difference.

Let me know if I can help further!
